### PR TITLE
Create/Save a new Friend Request

### PR DESCRIPTION
**Pull request by Shane-M-1**
I did this without the use of a DTO and instead, this returns a friend request directly. I didn't think a DTO would be necessary considering the simplicity of the FriendRequests model, but let me know if that should change!

### CODE REVIEW

File: src/main/java/com/danielagapov/spawn/Controllers/UserController.java [hunk 1]
```diff
@@ -51,4 +52,11 @@ public UserDTO createUser(@RequestBody UserDTO newUser) {
     public UserDTO replaceUser(@RequestBody UserDTO newUser, @PathVariable UUID id) {
         return userService.replaceUser(newUser, id);
     }
+
+    // full path: /api/v1/user/friendReq
+    @PostMapping("friendReq")
```
Comment (SeabertYuan) [inline src/main/java/com/danielagapov/spawn/Controllers/UserController.java:57]: Going off of the resource @Daggerpov sent on Slack, `/user/{id}/friend-requests` would follow REST URI conventions closer.

**Review by SeabertYuan — COMMENTED** [review]
From a design standpoint I think it's better to use DTO's for consistency but I think this is something we should discuss with @Daggerpov .

File: src/main/java/com/danielagapov/spawn/Controllers/UserController.java (hunk 1 above)
Comment (Shane-M-1) [reply src/main/java/com/danielagapov/spawn/Controllers/UserController.java:57]: Good point, I'll revisit that site.

File: src/main/java/com/danielagapov/spawn/Services/User/IUserService.java [hunk 2]
```diff
@@ -11,4 +12,5 @@ public interface IUserService {
     public List<UserDTO> getUsersByTagId(UUID tagId);
     public UserDTO saveUser(UserDTO user);
     public UserDTO replaceUser(UserDTO user, UUID id);
+    public FriendRequests saveFriendRequest(FriendRequests friendRequest);
```
Comment (Daggerpov) [inline src/main/java/com/danielagapov/spawn/Services/User/IUserService.java:15]: Please make a DTO, as discussed on Slack, that can simply inherit (`extends`) from `FriendRequests` for now.

File: src/main/java/com/danielagapov/spawn/Services/User/UserService.java [hunk 3]
```diff
@@ -70,4 +74,13 @@ public UserDTO replaceUser(UserDTO newUser, UUID id) {
             return UserMapper.toDTO(userEntity);
         });
     }
+
+    public FriendRequests saveFriendRequest(FriendRequests friendRequest) {
+        try {
+            friendRequestsRepository.save(friendRequest);
+            return friendRequest;
```
Comment (Daggerpov) [inline src/main/java/com/danielagapov/spawn/Services/User/UserService.java:81]: Once you make the entity -> DTO change, please save the entity to the database, and return the DTO to the controller (for the client [mobile]).

**Review by Daggerpov — CHANGES_REQUESTED** [review]
Great work. Just a couple of comments regarding the use of DTOs.

File: src/main/java/com/danielagapov/spawn/DTOs/FriendRequestDTO.java [hunk 4]
```diff
@@ -0,0 +1,9 @@
+package com.danielagapov.spawn.DTOs;
+
+import com.danielagapov.spawn.Models.User;
+
+import java.io.Serializable;
+import java.util.UUID;
+
+public record FriendRequestDTO(UUID id, User sender, User receiver) implements Serializable {
```
Comment (Daggerpov) [inline src/main/java/com/danielagapov/spawn/DTOs/FriendRequestDTO.java:8]: Hey, looks great! Just one thing, could you switch these `User` objects to `UserDTO`? Apart from that, everything else looks completely solid, so I'll approve right afterwards.

**Review by Daggerpov — CHANGES_REQUESTED** [review]
Looks great! Just `User` -> `UserDTO` in the `FriendRequestDTO`

**Review by Daggerpov — APPROVED** [review]

**Pull request merged**